api/src/main/java/jakarta/config/spi/ConfigSource.java

```diff
@@ -74,4 +76,24 @@
      * @return a set of property keys that are known to this configuration source
      */
     Set<String> getKeys();
+
+    /**
+     * Return {@code true} if this config source supports mutability (change of values in time).
+     * This method returns {@code false} by default.
+     *
+     * @return {@code true} for mutable config sources, {@code false} for immutable config sources
+     */
+    default boolean isMutable() {
```

Comment:
Do we need this method at all? If the onChange() is override, it means this config source is mutable. Otherwise, it is not.

Reply:
I am kind of thinking treating all config source the same and blidely call onChange()... for static config sources, the call will be empty call. anyway, just an idea.